staging/src/k8s.io/apiserver/pkg/storageversion/updater.go

```diff
@@ -35,23 +35,96 @@ type Client interface {
 	Get(context.Context, string, metav1.GetOptions) (*v1alpha1.StorageVersion, error)
 }
 
-func setCommonEncodingVersion(sv *v1alpha1.StorageVersion) {
+// SetCommonEncodingVersion updates the CommonEncodingVersion and the AllEncodingVersionsEqual
+// condition based on the StorageVersions.
+func SetCommonEncodingVersion(sv *v1alpha1.StorageVersion) {
+	var oldCommonEncodingVersion *string
+	if sv.Status.CommonEncodingVersion != nil {
+		version := *sv.Status.CommonEncodingVersion
+		oldCommonEncodingVersion = &version
+
+	}
 	if len(sv.Status.StorageVersions) == 0 {
+		sv.Status.CommonEncodingVersion = nil
+	} else {
+		firstVersion := sv.Status.StorageVersions[0].EncodingVersion
+		agreed := true
+		for _, ssv := range sv.Status.StorageVersions {
+			if ssv.EncodingVersion != firstVersion {
+				agreed = false
+				break
+			}
+		}
+		if agreed {
+			sv.Status.CommonEncodingVersion = &firstVersion
+		} else {
+			sv.Status.CommonEncodingVersion = nil
+		}
+	}
+
+	condition := v1alpha1.StorageVersionCondition{
+		Type:               v1alpha1.AllEncodingVersionsEqual,
+		Status:             v1alpha1.ConditionFalse,
+		ObservedGeneration: sv.Generation,
+		LastTransitionTime: metav1.NewTime(time.Now()),
+		Reason:             "CommonEncodingVersionUnset",
+		Message:            "Common encoding version unset",
```

Comment:
for free-form text we are likely to tweak in future versions, make sure we won't eternally duel with a future API server that sets a slightly different version of this condition. xref things like https://github.com/kubernetes/kubernetes/pull/78133